web/src/components/users/AdminUserEditForm.vue

```diff
@@ -0,0 +1,311 @@
+          :disabled="isNil(groupMembershipAttributes.branchId)"
+          @update:model-value="updateUnit"
+        />
+      </v-col>
+    </v-row>
+
+    <v-row>
+      <v-col
+        cols="12"
+        md="6"
+      >
+        <h3 class="mb-1">Roles</h3>
+
+        <v-chip
+          v-for="(roleType, index) in user.roleTypes"
+          :key="index"
+          class="ma-2"
+          color="info"
+          size="large"
+        >
+          {{ formatRole(roleType) }}
+        </v-chip>
+      </v-col>
+    </v-row>
+    <v-row>
+      <v-spacer />
+      <v-col class="d-flex justify-end">
+        <v-btn
+          :loading="isLoading"
+          color="error"
+          variant="outlined"
+          v-bind="cancelButtonOptions"
+          text="Cancel"
+        />
+        <!-- TODO: add ability to request update of yukon government directory information during save -->
+        <v-btn
+          class="ml-3"
+          :loading="isLoading"
+          type="submit"
+          color="success"
+          text="Save"
+        />
+      </v-col>
+    </v-row>
+  </v-form>
+</template>
+
+<script setup lang="ts">
+import { isNil } from "lodash"
+import { ref, toRefs, watch } from "vue"
+import { useI18n } from "vue-i18n"
+import { VBtn, VForm } from "vuetify/lib/components/index.mjs"
+
+import { required } from "@/utils/validators"
+
+import { GroupMembership } from "@/api/users-api"
+import { UserGroupTypes } from "@/api/user-groups-api"
+import useSnack from "@/use/use-snack"
+import useUser from "@/use/use-user"
+
+import UserGroupAutocomplete from "@/components/user-groups/UserGroupAutocomplete.vue"
+
+type CancelButtonOptions = VBtn["$props"]
+
+const props = withDefaults(
+  defineProps<{
+    userId: number
+    cancelButtonOptions?: CancelButtonOptions
+  }>(),
+  {
+    cancelButtonOptions: ({ userId }) => ({
+      to: {
+        name: "UserPage",
+        params: { userId },
+      },
+    }),
+  }
+)
+
+const emit = defineEmits(["saved"])
+
+const { userId } = toRefs(props)
+const { user, save, isLoading } = useUser(userId)
+
+const snack = useSnack()
+
+const groupMembershipAttributes = ref<Partial<GroupMembership>>({})
+
+const formRef = ref<InstanceType<typeof VForm> | null>(null)
+
+async function saveWrapper() {
+  if (formRef.value === null) return
+
+  const { valid } = await formRef.value.validate()
+  if (!valid) {
+    snack.notify("Please fill out all required fields", { color: "error" })
+    return
+  }
+
+  await save({
+    groupMembershipAttributes: groupMembershipAttributes.value,
+  })
+  emit("saved")
+}
+
+watch(
+  () => user.value,
+  (newUser) => {
+    if (isNil(newUser)) return
+
+    const { groupMembership } = newUser
+    const { departmentId, divisionId, branchId, unitId } = groupMembership
+
+    groupMembershipAttributes.value = {
+      departmentId,
+      divisionId,
+      branchId,
+      unitId,
+    }
+  },
+  {
+    deep: true,
+    immediate: true,
+  }
+)
```

Comment:
There's a bit of a UX bug around this code:
1. Set your department
2. Update you first name
3. Note that this resets your department.